### PR TITLE
Deprecate `observables` argument in `QubitDevice.statistics`

### PR DESCRIPTION
**Pull request by AlbertMitjans**
Add a deprecation warning when using `QubitDevice.statistics` with a list of observables, or when using the `observables` argument:
```pycon
>>> dev = qml.device("default.qubit", wires=0)
>>> dev.statistics([])
/Users/albertmitjans/Developer/pennylane/pennylane/_qubit_device.py:724: UserWarning: The ``observables`` argument in ``QubitDevice.statistics`` is deprecated. Please use ``circuit`` instead.
  warnings.warn(
>>> dev.statistics(observables=[])
/Users/albertmitjans/Developer/pennylane/pennylane/_qubit_device.py:724: UserWarning: The ``observables`` argument in ``QubitDevice.statistics`` is deprecated. Please use ``circuit`` instead.
  warnings.warn(
```

### CODE REVIEW

File: pennylane/_qubit_device.py
Comment (antalszava) [inline pennylane/_qubit_device.py:681]:
```suggestion
            circuit (~.tape.QuantumScript): the quantum script currently being executed
```

File: pennylane/_qubit_device.py
Comment (antalszava) [inline pennylane/_qubit_device.py:720]: Could this be a tape too?

File: pennylane/_qubit_device.py
Comment (antalszava) [inline pennylane/_qubit_device.py:855]:
```suggestion
            circuit (~.tape.QuantumScript): the quantum script currently being executed
```

**Review by antalszava — APPROVED** [review]
Looks good! :tada: Thanks for adding this.

**Comment by AlbertMitjans** [comment]
@trbromley @josh146 waiting for your approval before merging

**Pull request merged**
